src/style/selection_list.rs

```diff
@@ -1,34 +1,62 @@
-//! Selection List Redirects a Menu Style.
-use iced_style::menu;
+//! Selection List
+//!
+//! *This API requires the following crate features to be activated: `selection_list`*
+use iced_native::{Background, Color, Length};
+
+/// The appearance of a menu.
+#[derive(Debug, Clone, Copy)]
+pub struct Style {
+    /// The List Label Text Color
+    pub text_color: Color,
+    /// The background
+    pub background: Background,
+    /// The container Border width
+    pub border_width: f32,
+    /// The container Border color
+    pub border_color: Color,
+    /// The List Label Text Select Color
+    pub selected_text_color: Color,
+    /// The List Label Text Select Background Color
+    pub selected_background: Background,
+    /// The Containers Width
+    pub width: Length,
+    /// The Containers height
+    pub height: Length,
+    /// The padding Width
+    pub padding: u16,
+    /// The Text Size
+    pub text_size: u16,
```

Comment:
This is a little unusual. Normally, the style of the widgets do not change the behavior of the layout. You could place this fields in the struct of the widget and implement setters to modify this values.

Take a look at the [Column](https://github.com/hecrj/iced/blob/master/native/src/widget/column.rs) widget as an example.

Reply:
the issue was the only way I could make it work without requiring List to be a overlay was to do 

https://github.com/iced-rs/iced_aw/blob/c69a4a906fdfb9892403ca37edbcade59b38aea4/src/native/selection_list.rs#L75

which means I either need to pass the width and height and such in Directly OR pass it indirectly. In which this case the Style stuff was designed to be similar to that of CSS or at least I Thought. This seemed to be the better way of handling it since i do not get direct access into the containers children after they are created. So i cant actually Set these for the list except at the very start of the new function.


If you prefer it being in the new function Or see another way to handle it please let me know and I will fix the issue.

Reply:
Yes, this is indeed a little problem. Since I don't have a good solution for it at the moment either, we'll keep it for now and deal with it later.